File: core/distributed/distributed_execution_coordinator.py
from __future__ import annotations

from typing import Any, Dict, List


def coordinate_distributed_execution(
    schedules: List[Dict[str, Any]],
) -> Dict[str, Any]:

    ordered = sorted(
        schedules,
        key=lambda x: str(x),
    )

    return {
        "coordinated": ordered,
        "count": len(ordered),
    }

File: src/distributed/distributedExecutionCoordinator.ts
/**
 * Converted from Python: core/distributed/distributed_execution_coordinator.py
 * @generated — WebWeaveX python→javascript library port
 */

import * as py from "../runtime/pyCompat.js";

export function coordinateDistributedExecution(schedules: any): any {
  var ordered: any = py.sorted(schedules, {key: ((x: any) => py.toStr(x)) as (item: any) => any});
  return {"coordinated": ordered, "count": py.len(ordered)};
}
